identity/src/main/java/com/android/identity/mdoc/mso/StaticAuthDataGenerator.java

```diff
@@ -0,0 +1,170 @@
+ */
+public class StaticAuthDataGenerator {
+
+    private Map<String, List<byte[]>> mDigestIDMapping = new HashMap<>();
+    private byte[] mEncodedIssuerAuth;
+
+    /**
+     * Constructs a new {@link StaticAuthDataGenerator}.
+     */
+    public StaticAuthDataGenerator() {
+    }
+
+    /**
+     * Sets the <code>DigestIdMapping</code>. This must be called at least once before
+     * generating since <code>DigestIdMapping</code> must be non-empty.
+     *
+     * @param digestIDMapping A non-empty mapping between a <code>Namespace</code> and a list of
+     *                  <code>IssuerSignedItemBytes</code>.
+     * @return The <code>StaticAuthDataGenerator</code>.
+     * @exception IllegalArgumentException if the <code>digestIDMapping</code> is empty.
+     */
+    @NonNull
+    public StaticAuthDataGenerator setDigestIdMapping(@NonNull Map<String, List<byte[]>> digestIDMapping) {
+
+        if (digestIDMapping.isEmpty()) {
+            throw new IllegalArgumentException("digestIDs must not be empty");
+        }
+        mDigestIDMapping = digestIDMapping;
+        return this;
+    }
+
+    /**
+     * Populates the <code>issuerAuth</code> portion of the <code>StaticAuthData</code>. This
+     * contains the bytes of the MobileSecurityObject.
+     *
+     * @param encodedIssuerAuth a COSE_Sign1 object with a payload of MobileSecurityObjectBytes
+     * @return The <code>StaticAuthDataGenerator</code>.
+     */
+    @NonNull
+    public StaticAuthDataGenerator setEncodedIssuerAuth(@NonNull byte[] encodedIssuerAuth) {
+        mEncodedIssuerAuth = encodedIssuerAuth;
+        return this;
+    }
+
+    /**
+     * Builds the <code>StaticAuthData</code> CBOR.
+     *
+     * <p>The resulting byte array can be stored as <code>staticAuthData</code> using
+     * <code>IdentityCredential.storeStaticAuthenticationData({@link X509Certificate},
+     * {@link Calendar}, byte[])</code> and returned using
+     * <code>ResultData.getStaticAuthenticationData()}</code> at presentation time.
+     *
+     * @return the bytes of <code>StaticAuthData</code> CBOR.
+     * @exception IllegalStateException if required data hasn't been set using the setter
+     * methods on this class.
+     */
+    @NonNull
+    public byte[] generate() {
+        if (mEncodedIssuerAuth == null) {
+            throw new IllegalStateException("Must call setEncodedIssuerAuth before generating");
+        } else if (mDigestIDMapping.isEmpty()) {
+            throw new IllegalStateException("Must call setDigestIdMapping before generating");
+        }
+
+        CborBuilder digestIdBuilder = new CborBuilder();
+        MapBuilder<CborBuilder> outerBuilder = digestIdBuilder.addMap();
+        for (String namespace : mDigestIDMapping.keySet()) {
+            ArrayBuilder<MapBuilder<CborBuilder>> innerBuilder = outerBuilder.putArray(namespace);
+
+            for (byte[] encodedIssuerSignedItem : mDigestIDMapping.get(namespace)) {
```

Comment:
The docs say that `digestIDMapping` is "_A non-empty mapping between a `Namespace` and a list of `IssuerSignedItemBytes`_" which is indeed what we want.

But this code treats it as if it's a mapping of `IssuerSignedItem` e.g. without the `bstr` encoding. You can use `Util.cborExtractTaggedCbor()` for this.